WordPress/gutenberg
# Block Library: Deprecate the Post Comment block

## Description


Now that Comments Query Loop and Comment Template blocks are up and running we can finally deprecate the Post Comment block. The procedure is similar to what we did to the Text Columns block in the past:
- hide the block from the inserter
- update the title and description of the block to reflect the deprecation 

In this case, we might be able to remove the block completely after a few Gutenberg plugin releases because this block never went out of the experimental phase.

In addition to that, I update all blocks to use the Comment Template block as a template. I also changed the title, description, and category of several blocks to align with how Query Loop related blocks look like.

Bonus refactoring is based on the comment from @michalczaplinski in https://github.com/WordPress/gutenberg/pull/35965#discussion_r737734840 about the redundant `className` usage in `edit` method. It's handled with `useBlockProps` so we can remove it.

## How has this been tested?




1. Open the Site Editor for a block theme.
2. Navigate to the Single Post template.
3. Ensure you can't find the Post Comment block in the inserter. You should be able to insert it manually in the code editing mode when enabled.
4. Insert a Comments Query Loop block.
5. Insert more blocks inside the Comment Template block
    1. Comment Author Avatar.
    2. Comment Edit Link.
    3. Comment Reply Link. 
6. Ensure that the comments are rendered on the frontend after saving the template. 

## Screenshots 

All blocks in action inside the template:
https://user-images.githubusercontent.com/699132/139238006-d2d2610a-4023-4c5e-8019-e01dba733f36.mov

HTML inside the block editor (it contains all custom class names applied):
<img width="1038" alt="Screen Shot 2021-10-28 at 11 31 33" src="https://user-images.githubusercontent.com/699132/139238075-f3ad81e5-a2bd-4fe6-bc52-7c0ea16a5c6b.png">

HTML on the frontend (it contains all custom class names applied):
<img width="1280" alt="Screen Shot 2021-10-28 at 11 31 45" src="https://user-images.githubusercontent.com/699132/139238078-4ac18704-895c-4f70-899f-7b04d0d21355.png">

Searching in the inserter inside the Comment Template block:
<img width="338" alt="Screen Shot 2021-10-28 at 12 08 51" src="https://user-images.githubusercontent.com/699132/139238080-c8dc4852-178d-471d-a745-e3741d435ac9.png">

Searching in the inserter outside of the Comment Template block:
<img width="375" alt="Screen Shot 2021-10-28 at 12 11 03" src="https://user-images.githubusercontent.com/699132/139238081-1f0589e9-7177-4c5d-be23-a8b5d6114a44.png">



## Types of changes





## Checklist:
